src/ipc/ipc4/handler.c

```diff
@@ -369,7 +369,7 @@ static int ipc4_set_pipeline_state(union ipc4_message_header *ipc4)
 	struct ipc4_pipeline_set_state state;
 	uint32_t cmd, ppl_count;
 	uint32_t *ppl_id, id;
-	int ret;
+	int ret = 0;
```

Comment:
personally, I'd just use `return 0;` at line 402 and even moved the `ret` variable into the scope where it's used - inside the loop

Reply:
that would require modifying the logic at 399 as well since it drops its return value down.